physical/consul.go

```diff
@@ -527,16 +533,21 @@ func (c *ConsulBackend) reconcileConsul(registeredServiceID string, activeFunc a
 	sealed := sealedFunc()
 
 	agent := c.client.Agent()
+	catalog := c.client.Catalog()
+
+	serviceID = c.serviceID()
 
 	// Get the current state of Vault from Consul
-	var currentVaultService *api.AgentService
-	if services, err := agent.Services(); err == nil {
-		if service, ok := services[c.serviceName]; ok {
-			currentVaultService = service
+	var currentVaultService *api.CatalogService
+	if services, _, err := catalog.Service("vault", "", nil); err == nil {
```

Comment:
Since this goes to the servers and we are polling, it would be good to pass a `QueryOptions` with `AllowStale` set to `true`. This lets any server handle the request, which is more efficient, and prevents spurious errors during leader changes.